core/src/main/java/com/datastax/oss/driver/internal/core/channel/InFlightHandler.java

```diff
@@ -117,17 +117,24 @@ public void write(ChannelHandlerContext ctx, Object in, ChannelPromise promise)
   private void write(ChannelHandlerContext ctx, RequestMessage message, ChannelPromise promise) {
     if (closingGracefully) {
       promise.setFailure(new IllegalStateException("Channel is closing"));
+      streamIds.cancelPreAcquire();
       return;
     }
     int streamId = streamIds.acquire();
     if (streamId < 0) {
-      promise.setFailure(new BusyConnectionException(streamIds.getMaxAvailableIds()));
+      // Should not happen with the preAcquire mechanism, but handle gracefully
+      promise.setFailure(
+          new BusyConnectionException(
```

Comment:
don't we want to include this `streamIds.getMaxAvailableIds()` in the failure any more? Why?

Reply:
It's not easily obtainable from other places where the constructor is now called, and I thought it was pretty clear from the configuration.